examples/graphs/proteins_pytorch_geometric/model_def.py

```diff
@@ -56,6 +57,20 @@ def forward(self, data):
         return x
 
 
+def download_data_with_retry(n_retries, download_directory, dataset_name):
+    while n_retries > 0:
+        try:
+            return TUDataset(
+                root=download_directory,
+                name=dataset_name
+            )
+        except Exception as e:
+            n_retries -= 1
+            if n_retries==0:
+                traceback.print_exc()
+    raise Exception("Failed to download data.")
```

Comment:
I'd prefer we don't raise a bare `Exception`, a `RuntimeError` or even a `ValueError` is preferable I think.